module/skill/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,382 @@
+        cfg.arm_positions.emplace_back(ServoID::R_ELBOW, config["arms"]["right_elbow"].as<float>());
+        cfg.arm_positions.emplace_back(ServoID::L_ELBOW, config["arms"]["left_elbow"].as<float>());
+    }
+
+    QuinticWalk::QuinticWalk(std::unique_ptr<NUClear::Environment> environment)
+        : BehaviourReactor(std::move(environment)) {
+
+        imu_reaction = on<Trigger<Sensors>>().then([this](const Sensors& sensors) {
+            Eigen::Vector3f RPY =
+                utility::math::euler::MatrixToEulerIntrinsic(sensors.Htw.topLeftCorner<3, 3>().cast<float>());
+
+            // compute the pitch offset to the currently wanted pitch of the engine
+            float wanted_pitch =
+                current_cfg.params.trunk_pitch
+                + current_cfg.params.trunk_pitch_p_coef_forward * walk_engine.get_footstep().get_next().x()
+                + current_cfg.params.trunk_pitch_p_coef_turn * std::abs(walk_engine.get_footstep().get_next().z());
+            RPY.y() += wanted_pitch;
+
+            // threshold pitch and roll
+            if (std::abs(RPY.x()) > current_cfg.imu_roll_threshold) {
+                log<NUClear::WARN>(fmt::format("Robot roll exceeds threshold - {} > {}",
+                                               std::abs(RPY.x()),
+                                               current_cfg.imu_roll_threshold));
+                walk_engine.request_pause();
+            }
+            else if (std::abs(RPY.y()) > current_cfg.imu_pitch_threshold) {
+                log<NUClear::WARN>(fmt::format("Robot pitch exceeds threshold - {} > {}",
+                                               std::abs(RPY.y()),
+                                               current_cfg.imu_pitch_threshold));
+                walk_engine.request_pause();
+            }
+        });
+
+        on<Configuration>("QuinticWalk.yaml").then([this](const Configuration& config) {
+            log_level = config["log_level"].as<NUClear::LogLevel>();
+
+            load_quintic_walk(config, normal_config);
+
+            // Make sure the walk engine has the parameters at least once
+            if (first_config) {
+                // Send these parameters to the walk engine
+                walk_engine.set_parameters(current_cfg.params);
+
+                imu_reaction.enable(current_cfg.imu_active);
+
+                first_config = false;
+            }
+        });
+
+        on<Configuration>("goalie/QuinticWalk.yaml").then([this](const Configuration& config) {
+            load_quintic_walk(config, goalie_config);
+        });
+
+        on<Startup, Trigger<KinematicsModel>>().then("Update Kinematics Model", [this](const KinematicsModel& model) {
+            kinematicsModel = model;
+            first_run       = true;
+            current_orders.setZero();
+            is_left_support  = true;
+            last_update_time = NUClear::clock::now();
+            walk_engine.reset();
```

Comment:
move to `Start<Walk>` and get rid of this reaction. KinematicsModel can also be a `With<KinematicsModel>` in the main reactions where it's used.